### PR TITLE
Language switcher

### PR DESCRIPTION
**Pull request by Betsimisaraka**
Hello team,

This branch toggles language switcher button to switch language in every screen from English to Malagasy or vice versa.

And then fixed bug to enable us to use async storage by removing the storybook.

Thank you

https://app.clickup.com/t/35bjbb

### CODE REVIEW

File: src/screens/Home.js [hunk 1]
```diff
@@ -30,11 +30,14 @@ export default ({
   //state props
   categories,
   nativeLanguage,
+  secondLanguage,
   //actions
   setCategories,
   setCurrentCategory,
   setPhrases,
 }) => {
+  const [secondLanguages, setSecondLanguages] = useState(false);
```
Comment (andrzej-onja) [inline src/screens/Home.js]: We should avoid using hooks this should be added in HomeContainer and passed as a prop

File: src/screens/Home.js [hunk 2]
```diff
@@ -64,13 +67,15 @@ export default ({
             <ToolBar
               button={
                 <LanguageSwitcher
-                  firstLanguage={LANGUAGE_NAMES.EN}
-                  LeftText="EN"
-                  RightText="MA"
+                  firstLanguage={
+                    secondLanguages ? LANGUAGE_NAMES.MG : LANGUAGE_NAMES.EN
```
Comment (andrzej-onja) [inline src/screens/Home.js]: this condition is incorrect if secondlanguage is truthy its always LANGUAGE_NAMES.MG

File: src/redux/state.js [hunk 3]
```diff
@@ -5,4 +5,5 @@ export default {
   currentCategoryId: '',
   categoryPhrases: [],
   nativeLanguage: LANGUAGE_NAMES.EN,
+  secondLanguage: LANGUAGE_NAMES.MG,
```
Comment (andrzej-onja) [inline src/redux/state.js]: We don't need secondLanguage (please remove it) in state. We can derive second language just by checking native language. We only have to languages. 
if native is MG the second must be EN

File: src/screens/Learning.js [hunk 4]
```diff
@@ -112,13 +115,15 @@ export default ({
...
+                  firstLanguage={
+                    !secondLanguages ? LANGUAGE_NAMES.MG : LANGUAGE_NAMES.EN
+                  }
+                  LeftText={secondLanguages ? 'EN' : 'MA'}
+                  RightText={!secondLanguages ? 'EN' : 'MA'}
                   color="#FFFFFF"
                   iconType=""
                   iconName="swap-horiz"
-                  onPress={() => null}
+                  onPress={() => setSecondLanguages(!secondLanguages)}
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: This should also dispatch action to change value in redux store so every component will have access to value of current language

File: src/redux/actions/index.js [hunk 5]
```diff
@@ -34,3 +35,10 @@ export function setLanguageName(language) {
     payload: language,
   };
 }
+
+export function setSecondLanguage(language) {
+  return {
+    type: SET_SECOND_LANGUAGE,
+    payload: language,
+  };
+}
```
Comment (andrzej-onja) [inline src/redux/actions/index.js:137]: remove this and use existing setLanguageName then read the value of it in any component to decide on what lang version should be displayed

File: src/redux/selectors/index.js [hunk 6]
```diff
@@ -30,3 +31,11 @@ export const currentCategoryName = createSelector(
...
   },
 );
+
+export const secondCurrentCategoryName = createSelector(
+  [currentCategory, secondLanguageRoot],
+  (selectedCategory, language) => {
+    const name = selectedCategory?.name?.[language];
+    return name;
+  },
+);
```
Comment (andrzej-onja) [inline src/redux/selectors/index.js]: this is not needed

File: src/screens/Home.js [hunk 7]
```diff
@@ -98,34 +103,52 @@ export default ({
             />
           </View>
           <View style={styles.heading}>
-            <SectionHeading text="Select a category:" />
+            <SectionHeading
+              text={
+                secondLanguages ? 'Misafidiana sokajy:' : 'Select a category:'
+              }
+            />
```
Comment (andrzej-onja) [inline src/screens/Home.js]: Please create a new file with translation object that will be used in whole app (`\translations\index.js`):



```
// string ids
export const SELECT_CAT = 'SELECT_CAT';
export const LEARN = 'LEARN';

export const LANG_DATA = {
  [SELECT_CAT]: {
    [LANGUAGE_NAMES.EN]: 'Select a category:',
    [LANGUAGE_NAMES.MG]: 'Misafidiana sokajy:',
  },
  [LEARN]: {
    [LANGUAGE_NAMES.EN]: 'Learn',
    [LANGUAGE_NAMES.MG]: 'Minatra',
  },
};

```
then use it like this :
create a const at the top if component for each translated string
pass lang prop from redux state that is equal to native lang selected
 ```

 const selectCatText = LANG_DATA[SELECT_CAT][lang];
  const learnText = LANG_DATA[LEARN][lang];

```

Comment (andrzej-onja) [reply src/screens/Home.js]: repeat same approach for all strings inside the app that needs translations

File: src/redux/selectors/index.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ export const categoryPhrasesRoot = state => {
 };
 export const randomPhrase = state => state.randomPhrase;
 export const currentCategoryIdRoot = state => state.currentCategoryId;
+export const secondLanguageRoot = state => state.secondLanguage;
```
Comment (andrzej-onja) [inline src/redux/selectors/index.js]: We dont need second language

File: src/screens/Learning.js [hunk 9]
```diff
@@ -113,12 +143,12 @@ export default ({
...
                   firstLanguage={LANGUAGE_NAMES.EN}
-                  LeftText="EN"
-                  RightText="MA"
+                  LeftText={leftText}
+                  RightText={rightText}
                   color="#FFFFFF"
                   iconType=""
                   iconName="swap-horiz"
-                  onPress={() => null}
+                  onPress={() => toggleLanguageName()}
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: should be able to pass just function without calling it
onPress={toggleLanguageName}

File: src/redux/selectors/index.js (hunk 8 above)
Comment (andrzej-onja) [reply src/redux/selectors/index.js]: Also it seems to be unused

**Review by lightme-fan — APPROVED** [review]
Looks good, now!

**Review by ravinalamada — APPROVED** [review]

**Review by Tiaraha-funny — APPROVED** [review]

File: src/redux/actions/index.js (hunk 5 above)
Comment (constantingoeldel) [reply src/redux/actions/index.js:137]: Why was this resolved?

File: src/screens/Home.js [hunk 10]
```diff
@@ -109,41 +129,41 @@ export default ({
...
             makeAction={openCategoryPhrases}
           />
           <View style={styles.heading}>
-            <SectionHeading text="Seen phrases:" />
+            <SectionHeading text={seenPhraseText} />
           </View>
           <List
-            data={[{id: 1, name: '35 words and phrases'}]}
-            text={'Learn'}
+            data={[{id: 1, name: `35 ${wordsAndPhrases}`}]}
```
Comment (constantingoeldel) [inline src/screens/Home.js]: Is this really necessary?

File: yarn.lock [hunk 11]
```diff
@@ -1422,10 +1422,10 @@
     prop-types "^15.6.1"
     react-lifecycles-compat "^3.0.4"
 
-"@react-native-async-storage/async-storage@^1.15.4":
-  version "1.15.4"
-  resolved "http://192.168.1.46:4873/@react-native-async-storage%2fasync-storage/-/async-storage-1.15.4.tgz#cdba464ca3bb9f10ec538342cbf2520c06f453ab"
-  integrity sha512-pC0MS6UBuv/YiVAxtzi7CgUed8oCQNYMtGt0yb/I9fI/BWTiJK5cj4YtW2XtL95K5IuvPX/6uGWaouZ8KqXwdg==
+"@react-native-async-storage/async-storage@^1.15.5":
```
Comment (constantingoeldel) [inline yarn.lock]: We don't want to have the yarn.lock in source control

**Review by constantingoeldel — COMMENTED** [review]
Please talk to me or Andrzej about the choice to keep toggleLanguages in the code, otherwise it looks really good!

File: src/redux/actions/index.js (hunk 5 above)
Comment (Betsimisaraka) [reply src/redux/actions/index.js:137]: I have deleted these lines from my code. That's why I said it is resolved.

File: src/screens/Home.js (hunk 10 above)
Comment (Betsimisaraka) [reply src/screens/Home.js]: I have removed the repeated lines I just keep this one data={[{id: 1, name: `35 ${wordsAndPhrases}`}]}

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**